---
title: "Analysis of results: SBC on the GBM dataset from the whole Firebrowse Database(TCGA)."
subtitle: "AFT on the Oncogenic Gene Set "
author: "Camila Duitama (Based on Ashar Ahmad's code for the SBC model)"
date: "30th July 2019"
params:
  model: "OncogenicGeneSets+PAFT.RData"
  output_folder : "AnalysisOfResults_files/"

---
```{r setup, include=FALSE}
knitr::opts_chunk$set(cache=TRUE,include=TRUE,collapse=TRUE,warning=FALSE,message=FALSE, fig.width = 10, fig.height = 6)
rm(list = ls(all.names = TRUE))
load("/Volumes/GoogleDrive/My Drive/Documents/GitHub/MasterThesis/R_Scripts_First_Results/AnalysisOfResults.RData")
library(RColorBrewer)
library(limma)
VerhaakGeneSignature<-read_delim("/Volumes/GoogleDrive/My Drive/Documents/GitHub/Data/Verhaak_Data/Verhaak_gene_signature.csv", 
    ";", escape_double = FALSE, col_types = cols_only(`Gene Symbol` = col_guess()), 
    trim_ws = TRUE)
VerhaakGeneSignature<-VerhaakGeneSignature$`Gene Symbol`
```

# Description

This is the analysis of the SBC results after having fitting an AFT model the Oncogenic Gene Set downloaded from [MSigDB](http://software.broadinstitute.org/gsea/msigdb/genesets.jsp?collection=C6). A penalized AFT model was fit on the training data, and the linear predictors were then used as input features for the SBC model. This model (which was fit on the training data set) was then used to obtain linear predictor on the test set, which would be the test input features for the SBC.

# Loading of other datasets

```{r Load datasets, eval=FALSE}
Somatic_TrainingSet <- read_csv("../../Data/Second_Data_Split/TrainingSet/SomaticMutation_TrainingSet.csv")
CopyNumber_TrainingSet <- read_csv("../../Data/Second_Data_Split/TrainingSet/CopyNumber_TrainingSet.csv")
Somatic_TestSet <- read_csv("../../Data/Second_Data_Split/TestSet/SomaticMutation_TestSet.csv")
CopyNumber_TestSet <- read_csv("../../Data/Second_Data_Split/TestSet/CopyNumber_TestSet.csv")
```

# Copy Number Data

## Training set
```{r Chi-Square test for training set, eval=FALSE}
CopyNumber_TrainingSet$X1<-NULL
CopyNumber_TrainingSet$sbc.labels<-c.sbc
table<-list()
for (i in names(CopyNumber_TrainingSet)){
  if (i=="sbc.labels"){
    next
  }else{
  table[[i]]<-chisq.test(table(CopyNumber_TrainingSet$sbc.labels,CopyNumber_TrainingSet[[i]]))$p.value
  }
}
table.adj <- p.adjust(unlist(table), method = "fdr")
CN_pval <- names(table.adj[order(table.adj)[1:50]]) 
CN_signature<-CopyNumber_TrainingSet[CN_pval]

par(mar=c(1,1,1,1))

```

```{r heatmap training set copy number data}

CN_signature$sbc.labels<-c.sbc
CN_signature<-CN_signature[order(CN_signature$sbc.labels),]
ordered_labels<-CN_signature$sbc.labels
CN_signature$sbc.labels<-NULL
hmcols<-colorRampPalette(brewer.pal(11,"RdBu"))(7)
pdf(file="AnalysisOfResults_files/CopyNumber_TrainingSet.pdf")
heatmap.2(x = t(CN_signature), Rowv =TRUE ,Colv = FALSE, dendrogram = c("row"), scale ="row",ColSideColors =c("#b16ed3","#d37f6e","#90d36e","#6ec2d3")[ordered_labels], labRow = colnames(CN_signature), labCol = NA, main = ' \n Training Set \n Significant features for Copy Number Data', col =hmcols, cexRow =0.40, trace ="none", key.title = "Color Key")
legend("topright", legend = c("Good moderate","Worst","Best","Bad moderate"),fill =c("#b16ed3","#d37f6e","#90d36e","#6ec2d3"), cex = 0.4)
dev.off()
```

```{r}
hmcols<-colorRampPalette(brewer.pal(11,"RdBu"))(7)
heatmap.2(x = t(CN_signature), Rowv =TRUE ,Colv = FALSE, dendrogram = c("row"), scale ="row",ColSideColors =c("#b16ed3","#d37f6e","#90d36e","#6ec2d3")[ordered_labels], labRow = colnames(CN_signature), labCol = NA, main = ' \n Training Set \n Significant features \n Copy Number Data', col =hmcols, cexRow =0.40, trace ="none", key.title = "Color Key")
legend("topright", legend = c("Good moderate","Worst","Best","Bad moderate"),fill =c("#b16ed3","#d37f6e","#90d36e","#6ec2d3"), cex = 0.4)
```

## Test set

```{r Chi-Square test for test set, eval=FALSE}
CopyNumber_TestSet$X1<-NULL
CopyNumber_TestSet$sbc.labels<-c.sbc.new
table.new<-list()
for (i in names(CopyNumber_TestSet)){
  if (i=="sbc.labels"){
    next
  }else{
  table.new[[i]]<-chisq.test(table(CopyNumber_TestSet$sbc.labels,CopyNumber_TestSet[[i]]))$p.value
  }
}
table.adj.new <- p.adjust(unlist(table.new), method = "fdr")
CN_pval.new <- names(table.adj.new[order(table.adj.new)[1:50]]) 
CN_signature.new<-CopyNumber_TestSet[CN_pval.new]

par(mar=c(1,1,1,1))

```

```{r heatmap of CN for test set}
CN_signature.new$sbc.labels<-c.sbc.new
CN_signature.new<-CN_signature.new[order(CN_signature.new$sbc.labels),]
ordered_labels.new<-CN_signature.new$sbc.labels
CN_signature.new$sbc.labels<-NULL
hmcols<-brewer.pal(9,"RdBu")
heatmap.2(x = t(CN_signature.new), Rowv =TRUE ,Colv = FALSE, dendrogram = c("row"), scale ="row",ColSideColors =c("#b16ed3","#d37f6e","#90d36e","#6ec2d3")[ordered_labels.new], labRow = colnames(CN_signature.new), labCol = NA, main = ' \n Test Set \n Significant features \n Copy Number Data', col =hmcols, cexRow =0.40, trace ="none", key.title = "Color Key")
legend("topright", legend = c("Good moderate","Worst","Best","Bad moderate"),fill =c("#b16ed3","#d37f6e","#90d36e","#6ec2d3"), cex = 0.4)
```

# mRNA MicroArray Data

## Heatmap 

### Training Set

```{r Heatmap training set}
rank <- matrix(0, nrow = N, ncol =Nps)

for (j in 1:Nps){
  
  Y.scaled <- matrix(0, nrow = N, ncol =D)
  for ( v in 1:4){
    clust <- which(c.list[[j]] == v)
    Y.scaled[clust,1:D] <- scale(Y[clust,1:D], center = TRUE, scale = TRUE)
  }
  
  
  for ( i in 1:N){
    rank[i,j] <- dMVN(as.vector(t(Y[i,1:D])), mean = mu.list[[j]][c.final[i],1:D], Q= S.list[[j]][c.final[i],1:D,1:D], log = TRUE) - dMVN(as.vector(t(Y[i,1:D])), mean = mu.list[[j]][1,1:D], Q= S.list[[j]][1,1:D,1:D], log = TRUE) +  dnorm(x = That[i], mean = beta0.list[[j]][c.final[i]] + betahat.list[[j]][c.final[i],1:D] %*% as.vector(t(Y.scaled[i,1:D])), sd = sqrt(sigma2.list[[j]][c.final[i]]), log =TRUE) -  dnorm(x = That[i], mean = beta0.list[[j]][1] + betahat.list[[j]][1,1:D] %*% as.vector(t(Y.scaled[i,1:D])), sd = sqrt(sigma2.list[[j]][1]), log =TRUE) 
  }
}

avg.rank <- apply(rank,1,mean)
order.zo <- range01(avg.rank)

order.train <- sort(order.zo,index.return = TRUE, decreasing = TRUE)
Y.order <- Y[order.train$ix,]
c.final.order <- c.final[order.train$ix]


######## Reordering Again ################
order.2 <- c(which(c.final.order==1),which(c.final.order==3),which(c.final.order==4),which(c.final.order==2))
Y.order.2 <- Y.order[order.2,]
c.final.order.2 <- c.final.order[order.2]

hmcols<-colorRampPalette(brewer.pal(11,"RdBu"))(256)
pdf(file="AnalysisOfResults_files/SBC_signature_TrainingSet.pdf")
heatmap.2(x = t(Y.order.2), Rowv =TRUE ,Colv = FALSE, dendrogram = c("row"), scale ="row",ColSideColors =c("#b16ed3","#d37f6e","#90d36e","#6ec2d3")[c.final.order.2], labRow = colnames(Y.order.2), labCol = NA, main = ' \n Training Set \n Gene Sets SBC Signature', col =hmcols, cexRow =0.40, trace ="none", key.title = "Color Key")
legend("topright", legend = c("Good moderate","Worst","Best","Bad moderate"),fill =c("#b16ed3","#d37f6e","#90d36e","#6ec2d3"), cex = 0.4)
dev.off()

```

```{r}
hmcols<-colorRampPalette(brewer.pal(11,"RdBu"))(256)
heatmap.2(x = t(Y.order.2), Rowv =TRUE ,Colv = FALSE, dendrogram = c("row"), scale ="row",ColSideColors =c("#b16ed3","#d37f6e","#90d36e","#6ec2d3")[c.final.order.2], labRow = colnames(Y.order.2), labCol = NA, main = ' \n Training Set \n Gene Sets SBC Signature', col =hmcols, cexRow =0.40, trace ="none", key.title = "Color Key")
legend("topright", legend = c("Good moderate","Worst","Best","Bad moderate"),fill =c("#b16ed3","#d37f6e","#90d36e","#6ec2d3"), cex = 0.4)
```

### Test set

```{r}
rank <- matrix(0, nrow = N.new, ncol =Nps)

for (j in 1:Nps){
  
  Y.scaled <- matrix(0, nrow = N.new, ncol =D)
  for ( v in 1:4){
    clust <- which(c.matrix.new[,j] == v)
    Y.scaled[clust,1:D] <- scale(Y.new[clust,1:D], center = TRUE, scale = TRUE)
  }
  
  
  for ( i in 1:N.new){
    rank[i,j] <- dMVN(as.vector(t(Y.new[i,1:D])), mean = mu.list[[j]][c.sbc.new[i],1:D], Q= S.list[[j]][c.sbc.new[i],1:D,1:D], log = TRUE) - dMVN(as.vector(t(Y.new[i,1:D])), mean = mu.list[[j]][1,1:D], Q= S.list[[j]][1,1:D,1:D], log = TRUE) +  dnorm(x = time.new[i], mean = beta0.list[[j]][c.sbc.new[i]] + betahat.list[[j]][c.sbc.new[i],1:D] %*% as.vector(t(Y.scaled[i,1:D])), sd = sqrt(sigma2.list[[j]][c.sbc.new[i]]), log =TRUE) -  dnorm(x = time.new[i], mean = beta0.list[[j]][1] + betahat.list[[j]][1,1:D] %*% as.vector(t(Y.scaled[i,1:D])), sd = sqrt(sigma2.list[[j]][1]), log =TRUE) 
  }
}

avg.rank <- apply(rank,1,mean)
order.zo <- range01(avg.rank)

order.train <- sort(order.zo,index.return = TRUE, decreasing = TRUE)
Y.order.new <- Y.new[order.train$ix,]
c.final.order.new <- c.sbc.new[order.train$ix]


######## Reordering Again ################
order.2.new <- c(which(c.final.order.new==1),which(c.final.order.new==3),which(c.final.order.new==4),which(c.final.order.new==2))
Y.order.2.new <- Y.order.new[order.2.new,]
c.final.order.2.new <- c.final.order.new[order.2.new]

hmcols<-colorRampPalette(brewer.pal(11,"RdBu"))(256)
heatmap.2(x = t(Y.order.2.new), Rowv =TRUE ,Colv = FALSE, dendrogram = c("row"), scale ="row",ColSideColors =c("#b16ed3","#d37f6e","#90d36e","#6ec2d3")[c.final.order.2.new], labRow = colnames(Y.order.2.new), labCol = NA, main = ' \n Testing Set \n 47 Gene DPMM signature', col =hmcols, cexRow =0.40, trace ="none", key.title = "Color Key")
legend("topright", legend = c("Good moderate","Worst","Best","Bad moderate"),fill =c("#b16ed3","#d37f6e","#90d36e","#6ec2d3"), cex = 0.4)

```

## Differential expression

### Training set

#### Significant mRNA overall
```{r differential expression of training set}
##### Load the SBC Labels
cluster  <- factor(c.sbc)
##### Create the design matrix with these labels
mm <- model.matrix( ~0+cluster)

##### You can also define a contrast matrix which only looks at cluster specific contrasts
cont.matrix <- makeContrasts(Cluster1vsCluster2 = cluster1 - cluster2,Cluster1vsCluster3=cluster1 - cluster3,Cluster1vsCluster4=cluster1 - cluster4,Cluster2vsCluster3= cluster2 - cluster3,
Cluster2vsCluster4=cluster2 - cluster4,Cluster3vsCluster4=cluster3 - cluster4,levels= mm)
###### Fit limma model
fit <- lmFit(t(mRNAArray_TrainingSet), design = mm)
efit <- eBayes(fit)
###### Look for DE table
topmRNA.train<-rownames(topTable(efit,number=50,adjust.method="BH",p.val=0.05))

mRNA.matrix.train<-mRNAArray_TrainingSet[,topmRNA.train]
mRNA.matrix.train$sbc.labels<-c.sbc
mRNA.matrix.train<-mRNA.matrix.train[order(mRNA.matrix.train$sbc.labels),]
ordered_labels<-mRNA.matrix.train$sbc.labels
mRNA.matrix.train$sbc.labels<-NULL

hmcols<-colorRampPalette(brewer.pal(11,"RdBu"))(256)
pdf(file="AnalysisOfResults_files/DEmRNA_TrainingSet.pdf")
heatmap.2(x = t(mRNA.matrix.train),Colv = FALSE, Rowv=TRUE, dendrogram= c("row"), scale ="row",ColSideColors =c("#b16ed3","#d37f6e","#90d36e","#6ec2d3")[ordered_labels], labRow = colnames(mRNA.matrix.train), labCol = NA, main = ' \n Training Set \n DE mRNA transcripts', col =hmcols, cexRow =0.40, trace ="none", key.title = "Color Key")
legend("topright", legend = c("Good moderate","Worst","Best","Bad moderate"),fill =c("#b16ed3","#d37f6e","#90d36e","#6ec2d3"), cex = 0.4)
dev.off()

```

```{r}
heatmap.2(x = t(mRNA.matrix.train),Colv = FALSE, Rowv=TRUE, dendrogram= c("row"), scale ="row",ColSideColors =c("#b16ed3","#d37f6e","#90d36e","#6ec2d3")[ordered_labels], labRow = colnames(mRNA.matrix.train), labCol = NA, main = ' \n Training Set \n DE mRNA transcripts', col =hmcols, cexRow =0.40, trace ="none", key.title = "Color Key")
legend("topright", legend = c("Good moderate","Worst","Best","Bad moderate"),fill =c("#b16ed3","#d37f6e","#90d36e","#6ec2d3"), cex = 0.4)
```

#### Comparison between DE mRNA transcripts and SBC signature

```{r Comparsion between DE mRNA and SBC signature for the training}
intersect(signature.sbc,topmRNA.train)
```

#### Significant mRNA in pairwise cluster comparison
```{r significant mrna pairwise  of training set}
##### Load the SBC Labels
cluster  <- factor(c.sbc)
##### Create the design matrix with these labels
design<- model.matrix( ~0+cluster)
fit<-lmFit(t(mRNAArray_TrainingSet) , design= design)

##### You can also define a contrast matrix which only looks at cluster specific contrasts
cont.matrix <- makeContrasts(Cluster1vsCluster2 = cluster1 - cluster2,Cluster1vsCluster3=cluster1 - cluster3,Cluster1vsCluster4=cluster1 - cluster4,Cluster2vsCluster3= cluster2 - cluster3,
Cluster2vsCluster4=cluster2 - cluster4,Cluster3vsCluster4=cluster3 - cluster4,levels= design)

###### Fit limma model
fit.2 <- contrasts.fit(fit, contrasts = cont.matrix)
fit.2 <- eBayes(fit.2, trend= TRUE)
###### Look for DE table
topmRNApairwise.train<-rownames(topTable(fit.2,number=50, p.val=0.05))

mRNA.matrix.train<-mRNAArray_TrainingSet[,topmRNApairwise.train]
mRNA.matrix.train$sbc.labels<-c.sbc
mRNA.matrix.train<-mRNA.matrix.train[order(mRNA.matrix.train$sbc.labels),]
ordered_labels<-mRNA.matrix.train$sbc.labels
mRNA.matrix.train$sbc.labels<-NULL

#### Significant mRNA in pairwise cluster comparison
par(mar=c(1,1,1,1))
pairwise.mRNA.train<-as.matrix(topTable(fit.2,number = 20,adjust.method="BH",p.val=0.05)[,1:6])
hmcols<-colorRampPalette(brewer.pal(11,"RdBu"))(256)
pdf(file="AnalysisOfResults_files/DEmRNA_pairwise_TrainingSet.pdf")
heatmap.2(x = pairwise.mRNA.train,Colv = FALSE, Rowv=TRUE, dendrogram= c("row"), scale ="row", labRow =rownames(pairwise.mRNA.train), labCol = colnames(pairwise.mRNA.train), main ='\n Training set \n Pairwise cluster comparison \n of DE mRNA', col =hmcols, cexRow =0.40,cexCol = 0.7, trace ="none", key.title = "Color Key", margins=c(6,10))
dev.off()

```
```{r}
hmcols<-colorRampPalette(brewer.pal(11,"RdBu"))(256)
heatmap.2(x = pairwise.mRNA.train,Colv = FALSE, Rowv=TRUE, dendrogram= c("row"), scale ="row", labRow =rownames(pairwise.mRNA.train), labCol = colnames(pairwise.mRNA.train), main = ' \n Pairwise cluster comparison \n Significant mRNA for the Training Set', col =hmcols, cexRow =0.40,cexCol = 0.7, trace ="none", key.title = "Color Key", margins=c(6,10))
```




### Test set

```{r differential expression of test set}
##### Load the SBC Labels
cluster.new  <- factor(c.sbc.new)
##### Create the design matrix with these labels
mm.new <- model.matrix( ~0+cluster.new)

##### You can also define a contrast matrix which only looks at cluster specific contrasts
cont.matrix <- makeContrasts(Cluster1vsCluster2 = cluster1 - cluster2,Cluster1vsCluster3=cluster1 - cluster3,Cluster1vsCluster4=cluster1 - cluster4,Cluster2vsCluster3= cluster2 - cluster3,
Cluster2vsCluster4=cluster2 - cluster4,Cluster3vsCluster4=cluster3 - cluster4,levels= mm)
###### Fit limma model
fit.new <- lmFit(t(mRNAArray_TestSet), design = mm.new)
efit.new <- eBayes(fit.new)
###### Look for DE table
topmRNA.test<-rownames(topTable(efit.new,number=50,adjust.method="BH"))

mRNA.matrix.test<-mRNAArray_TestSet[,topmRNA.test]
mRNA.matrix.test$sbc.labels<-c.sbc.new
mRNA.matrix.test<-mRNA.matrix.test[order(mRNA.matrix.test$sbc.labels),]
ordered_labels.new<-mRNA.matrix.test$sbc.labels
mRNA.matrix.test$sbc.labels<-NULL

hmcols<-colorRampPalette(brewer.pal(11,"RdBu"))(256)
heatmap.2(x = t(mRNA.matrix.test),Colv = FALSE, Rowv=TRUE, dendrogram= c("row"), scale ="row",ColSideColors =c("#b16ed3","#d37f6e","#90d36e","#6ec2d3")[ordered_labels.new], labRow = colnames(mRNA.matrix.test), labCol = NA, main = ' \n Test Set \n Differentially expressed mRNA transcripts', col =hmcols, cexRow =0.40, trace ="none", key.title = "Color Key")
legend("topright", legend = c("Good moderate","Worst","Best","Bad moderate"),fill =c("#b16ed3","#d37f6e","#90d36e","#6ec2d3"), cex = 0.4)
```

#### Comparison between DE mRNA transcripts and SBC signature

```{r Comparsion between DE mRNA and SBC signature for the test}
intersect(signature.sbc,topmRNA.test)
```

# SBC Feature Importance

```{r Feature importance}
range01 <- function(x){(x-min(x))/(max(x)-min(x))}

##### Visualization of the Results ###############################
#### USE Top 35 Molecular Features OBTAINED FROM THE MODEL ############
sum.diff12  <- c(0)
for ( i in 1:Nps){
  sum.diff12 <- sum.diff12 + abs(mu.list[[i]][1,] - mu.list[[i]][2,])  
}
sum.diff12.sc <- (1/Nps) * (sum.diff12)/(diag(W))
cluste12 <- range01(sum.diff12.sc)

sum.diff23  <- c(0)
for ( i in 1:Nps){
  sum.diff23 <- sum.diff23 + abs(mu.list[[i]][2,] - mu.list[[i]][3,])  
}
sum.diff23.sc <- (1/Nps) * (sum.diff23)/(diag(W))

cluste23 <- range01(sum.diff23.sc)

sum.diff34  <- c(0)
for ( i in 1:Nps){
  sum.diff34 <- sum.diff34 + abs(mu.list[[i]][3,] - mu.list[[i]][4,])  
}
sum.diff34.sc <- (1/Nps) * (sum.diff34)/(diag(W))

cluste34 <- range01(sum.diff34.sc)

sum.diff41  <- c(0)
for ( i in 1:Nps){
  sum.diff41 <- sum.diff34 + abs(mu.list[[i]][4,] - mu.list[[i]][1,])  
}
sum.diff41.sc <- (1/Nps) * (sum.diff41)/(diag(W))

cluste41 <- range01(sum.diff41.sc)

sum.diff13  <- c(0)
for ( i in 1:Nps){
  sum.diff13 <- sum.diff13 + abs(mu.list[[i]][1,] - mu.list[[i]][3,])  
}
sum.diff13.sc <- (1/Nps) * (sum.diff13)/(diag(W))

cluste13 <- range01(sum.diff13.sc)

sum.diff24  <- c(0)
for ( i in 1:Nps){
  sum.diff24 <- sum.diff24 + abs(mu.list[[i]][2,] - mu.list[[i]][4,])  
}
sum.diff24.sc <- (1/Nps) * (sum.diff24)/(diag(W))
cluste24 <- range01(sum.diff24.sc)

####################### Feature Signficance matrix
cluste <- cbind(cluste12,cluste23,cluste34, cluste41, cluste13, cluste24)
heatmapdata <- cluste
rownames(heatmapdata) <-  colnames(Y)
colnames(heatmapdata) <- c("1 v s2","2 vs 3","3 v s4","4 vs 1","1 vs 3","2 vs 4")
hmcols<-colorRampPalette(c("white","black"))(128)
```

```{r}
cluste <- cbind(cluste12,cluste23,cluste34, cluste41, cluste13, cluste24)
heatmapdata <- cluste
rownames(heatmapdata) <-  colnames(Y)
colnames(heatmapdata) <- c("1 v s2","2 vs 3","3 v s4","4 vs 1","1 vs 3","2 vs 4")
hmcols<-colorRampPalette(c("white","black"))(128)
pdf("AnalysisOfResults_files/FeatureImportance.pdf")
heatmap.2(heatmapdata , margins=c(6,10),col = hmcols, main = "SBC signature \n Feature Importance \n GBM ", cexCol = 0.85, cexRow = 0.7, Rowv = TRUE, Colv = TRUE, trace = "none", xlab = "Clusters") 
dev.off()
```



# Association between Verhaak labels and SBC labels

```{r Association between Verhaak labels and SBC labels}
########## Look for Relationship between Our classes and Verhaak Classes ########################
table(c.sbc,c.true)
matrix.assoc <- rbind(c(29,33,18,25), c(1,1,1,0), c(3,8,2,15), c(3,9,3,9))
fisher.test(matrix.assoc, workspace = 10000000,hybrid =TRUE)
```

# miRNA Data

```{r Loading miRNA training, eval= FALSE}
library(multiMiR)
#Load data
miRNA_TrainingSet <- read_csv("../../Data/Second_data_split/TrainingSet/miRNA_TrainingSet.csv")
miRNA_TrainingSet$X1<-NULL
```

## Expression data

### Training set

```{r PCA of miRNA data for training}
miRNA.train.pca<-as.data.frame(prcomp(miRNA_TrainingSet,scale=TRUE, center= TRUE)$x[,1:2])
miRNA.train.pca$SBC_labels<-c.sbc
plotmiRNA.train <- ggplot(miRNA.train.pca, aes(x=miRNA.train.pca$PC1, y= miRNA.train.pca$PC2, colour= as.factor(SBC_labels))) + ggtitle("PCA \n Training set with SBC labels and unlabeled test set") + geom_point(shape=19) + labs(y = "PC1", x = "PC2", colour = "SBC_labels") +scale_color_manual(values=c("#b16ed3","#d37f6e","#90d36e","#6ec2d3","#858687"))
plotmiRNA.train
```

```{r Loading miRNA testing, eval= FALSE}
#Load data
miRNA_TestSet <- read_csv("/Volumes/GoogleDrive/My Drive/Documents/GitHub//Data/Second_data_split/TestSet/miRNA_TestSet.csv")
miRNA_TestSet$X1<-NULL
```


### Test set

```{r PCA of miRNA data for testing}
miRNA.test.pca<-as.data.frame(prcomp(miRNA_TestSet,scale=TRUE, center= TRUE)$x[,1:2])
miRNA.test.pca$SBC_labels<-c.sbc.new
plotmiRNA.test <- ggplot(miRNA.test.pca, aes(x=miRNA.test.pca$PC1, y= miRNA.test.pca$PC2, colour= as.factor(SBC_labels))) + ggtitle("PCA \n Training set with SBC labels and unlabeled test set") + geom_point(shape=19) + labs(y = "PC1", x = "PC2", colour = "SBC_labels") +scale_color_manual(values=c("#b16ed3","#d37f6e","#90d36e","#6ec2d3","#858687"))
plotmiRNA.test
```

## Validated targets 

### Training Set
```{r miRNA validated targets for the training set, eval=FALSE}

#Find all the validated targets for the miRNA present in our data
result<-list()
try(example1 <- get_multimir(mirna = names(miRNA_TrainingSet), summary = TRUE))
result <- unlist(unique(example1@data$target_symbol))
names.use <- names(mRNAArray_TrainingSet)[(names(mRNAArray_TrainingSet) %in% result)]

```

```{r PCA of the expression of the mRNA targets for the training set}

#PCA of the expression of the mRNA targets
targets.train<-mRNAArray_TrainingSet[,names.use]
targets.train.pca<-as.data.frame(prcomp(targets.train,scale=TRUE, center= TRUE)$x[,1:2])
targets.train.pca$SBC_labels<-c.sbc
p.targets.train <- ggplot(targets.train.pca, aes(x=targets.train.pca$PC1, y= targets.train.pca$PC2, colour= as.factor(SBC_labels))) + ggtitle("PCA \n Training set with SBC labels and unlabeled test set") + geom_point(shape=19) + labs(y = "PC1", x = "PC2", colour = "SBC_labels") +scale_color_manual(values=c("#b16ed3","#d37f6e","#90d36e","#6ec2d3","#858687"))
p.targets.train

```


### Test Set
```{r miRNA validated targets for the test set, eval=FALSE}

#Find all the validated targets for the miRNA present in our data
result.new<-list()
try(example1.new <- get_multimir(mirna = names(miRNA_TestSet), summary = TRUE))
result.new <- unlist(unique(example1.new@data$target_symbol))
names.use.new <- names(mRNAArray_TestSet)[(names(mRNAArray_TestSet) %in% result.new)]

```

```{r miRNA validated targets for the testing set}
#PCA of the expression of the mRNA targets
targets.test<-mRNAArray_TestSet[,names.use.new]
targets.test.pca<-as.data.frame(prcomp(targets.test,scale=TRUE, center= TRUE)$x[,1:2])
targets.test.pca$SBC_labels<-c.sbc.new
p.targets.test <- ggplot(targets.test.pca, aes(x=targets.test.pca$PC1, y= targets.test.pca$PC2, colour= as.factor(SBC_labels))) + ggtitle("PCA \n Training set with SBC labels and unlabeled test set") + geom_point(shape=19) + labs(y = "PC1", x = "PC2", colour = "SBC_labels") +scale_color_manual(values=c("#b16ed3","#d37f6e","#90d36e","#6ec2d3","#858687"))
p.targets.test
```

## Differential expression

### Training set

#### Significant miRNA overall
```{r miRNA differential expression  of training set}
##### Load the SBC Labels
cluster  <- factor(c.sbc)
##### Create the design matrix with these labels
design<- model.matrix( ~0+cluster)
fit<-lmFit(t(miRNA_TrainingSet) , design= design)

##### You can also define a contrast matrix which only looks at cluster specific contrasts
cont.matrix <- makeContrasts(Cluster1vsCluster2 = cluster1 - cluster2,Cluster1vsCluster3=cluster1 - cluster3,Cluster1vsCluster4=cluster1 - cluster4,Cluster2vsCluster3= cluster2 - cluster3,
Cluster2vsCluster4=cluster2 - cluster4,Cluster3vsCluster4=cluster3 - cluster4,levels= design)

###### Fit limma model
fit.2 <- contrasts.fit(fit, contrasts = cont.matrix)
fit.2 <- eBayes(fit.2, trend= TRUE)
###### Look for DE table
topmiRNA.train<-rownames(topTable(fit.2,number=50,p.val=0.05))

miRNA.matrix.train<-miRNA_TrainingSet[,topmiRNA.train]
miRNA.matrix.train$sbc.labels<-c.sbc
miRNA.matrix.train<-miRNA.matrix.train[order(miRNA.matrix.train$sbc.labels),]
ordered_labels<-miRNA.matrix.train$sbc.labels
miRNA.matrix.train$sbc.labels<-NULL

par(mar=c(1,1,1,1))

pdf(file="AnalysisOfResults_files/DEmiRNA_TrainingSet.pdf")
hmcols<-colorRampPalette(brewer.pal(11,"RdBu"))(256)
heatmap.2(x = t(miRNA.matrix.train),Colv = FALSE, Rowv=TRUE, dendrogram= c("row"), scale ="row",ColSideColors =c("#b16ed3","#d37f6e","#90d36e","#6ec2d3")[ordered_labels], labRow = colnames(miRNA.matrix.train), labCol = NA, main = ' \n Training Set \n DE miRNA transcripts', col =hmcols, cexRow =0.40, trace ="none", key.title = "Color Key")
legend("topright", legend = c("Good moderate","Worst","Best","Bad moderate"),fill =c("#b16ed3","#d37f6e","#90d36e","#6ec2d3"), cex = 0.4)
dev.off()

```

```{r}

hmcols<-colorRampPalette(brewer.pal(11,"RdBu"))(256)
heatmap.2(x = t(miRNA.matrix.train),Colv = FALSE, Rowv=TRUE, dendrogram= c("row"), scale ="row",ColSideColors =c("#b16ed3","#d37f6e","#90d36e","#6ec2d3")[ordered_labels], labRow = colnames(miRNA.matrix.train), labCol = NA, main = ' \n Training Set \n DE miRNA transcripts', col =hmcols, cexRow =0.40, trace ="none", key.title = "Color Key")
legend("topright", legend = c("Good moderate","Worst","Best","Bad moderate"),fill =c("#b16ed3","#d37f6e","#90d36e","#6ec2d3"), cex = 0.4)
```

#### Significant miRNA in pairwise cluster comparison

```{r Significant miRNA transcript in pairwise comparisons for training}
par(mar=c(1,1,1,1))
pairwise.miRNA.train<-as.matrix(topTable(fit.2,number = 20,adjust.method="BH",p.val=0.05)[,1:6])
pdf(file="AnalysisOfResults_files/DEmiRNA_pairwise_TrainingSet.pdf")
hmcols<-colorRampPalette(brewer.pal(11,"RdBu"))(256)
heatmap.2(x = pairwise.miRNA.train,Colv = FALSE, Rowv=TRUE, dendrogram= c("row"), scale ="row", labRow =rownames(pairwise.miRNA.train), labCol = colnames(pairwise.miRNA.train), main = ' \n Training set \n DE miRNA transcripts \n in cluster pairwise comparison', col =hmcols, cexRow =0.40,cexCol = 0.7, trace ="none", key.title = "Color Key", margins=c(6,10))
dev.off()
```

```{r}
hmcols<-colorRampPalette(brewer.pal(11,"RdBu"))(256)
heatmap.2(x = pairwise.miRNA.train,Colv = FALSE, Rowv=TRUE, dendrogram= c("row"), scale ="row", labRow =rownames(pairwise.miRNA.train), labCol = colnames(pairwise.miRNA.train), main = ' \n Pairwise cluster comparison \n Significant miRNA for the Training Set', col =hmcols, cexRow =0.40,cexCol = 0.7, trace ="none", key.title = "Color Key", margins=c(6,10))
```

#### PCA of differentially expressed miRNA targets

```{r PCA of differentially expressed miRNA for the training, eval=FALSE}
#Find all the validated targets
result.de.train<-list()
try(example2 <- get_multimir(mirna = topmiRNA.train, summary = TRUE))
result.de.train <- unlist(unique(example2@data$target_symbol))

tokeep<-colnames(mRNAArray_TrainingSet)[colnames(mRNAArray_TrainingSet) %in% result.de.train]
DEtargets.train.pca<-as.data.frame(prcomp(mRNAArray_TrainingSet[,tokeep],scale=TRUE, center= TRUE)$x[,1:2])
DEtargets.train.pca$SBC_labels<-c.sbc
plotDEtargets.train <- ggplot(miRNA.train.pca, aes(x=DEtargets.train.pca$PC1, y= DEtargets.train.pca$PC2, colour= as.factor(SBC_labels))) + ggtitle("PCA \n Validated DE miRNA targets \n Training Set") + geom_point(shape=19) + labs(y = "PC1", x = "PC2", colour = "SBC_labels") +scale_color_manual(values=c("#b16ed3","#d37f6e","#90d36e","#6ec2d3","#858687"))
plotDEtargets.train
```

### Test set

#### Significant miRNA overall

```{r miRNA differential expression  of test set}
##### Load the SBC Labels
cluster.new  <- factor(c.sbc.new)
##### Create the design matrix with these labels
design.new<- model.matrix( ~0+cluster.new)
fit.new<-lmFit(t(miRNA_TestSet) , design= design.new)

##### You can also define a contrast matrix which only looks at cluster specific contrasts
cont.matrix <- makeContrasts(Cluster1vsCluster2 = cluster1 - cluster2,Cluster1vsCluster3=cluster1 - cluster3,Cluster1vsCluster4=cluster1 - cluster4,Cluster2vsCluster3= cluster2 - cluster3,
Cluster2vsCluster4=cluster2 - cluster4,Cluster3vsCluster4=cluster3 - cluster4,levels= design)

###### Fit limma model
fit.2.new <- contrasts.fit(fit.new, contrasts = cont.matrix)
fit.2.new <- eBayes(fit.2.new, trend= TRUE)
###### Look for DE table
topmiRNA.test<-rownames(topTable(fit.2.new,number=50))

miRNA.matrix.test<-miRNA_TestSet[,topmiRNA.test]
miRNA.matrix.test$sbc.labels<-c.sbc.new
miRNA.matrix.test<-miRNA.matrix.test[order(miRNA.matrix.test$sbc.labels),]
ordered_labels<-miRNA.matrix.test$sbc.labels
miRNA.matrix.test$sbc.labels<-NULL

par(mar=c(1,1,1,1))

hmcols<-colorRampPalette(brewer.pal(11,"RdBu"))(256)
heatmap.2(x = t(miRNA.matrix.test),Colv = FALSE, Rowv=TRUE, dendrogram= c("row"), scale ="row",ColSideColors =c("#b16ed3","#d37f6e","#90d36e","#6ec2d3")[ordered_labels], labRow = colnames(miRNA.matrix.test), labCol = NA, main = ' \n Test Set \n Significant miRNA transcripts', col =hmcols, cexRow =0.40, trace ="none", key.title = "Color Key")
legend("topright", legend = c("Good moderate","Worst","Best","Bad moderate"),fill =c("#b16ed3","#d37f6e","#90d36e","#6ec2d3"), cex = 0.4)
```

#### Significant miRNA in pairwise cluster comparison

```{r Significant miRNA transcript in pairwise comparisons for testing}
par(mar=c(1,1,1,1))
pairwise.miRNA.test<-as.matrix(topTable(fit.2.new,number = 20,adjust.method="BH")[,1:6])
hmcols<-colorRampPalette(brewer.pal(11,"RdBu"))(256)
pdf(file="AnalysisOfResults_files/DEmiRNA_pairwise_TestSet.pdf")
heatmap.2(x = pairwise.miRNA.test,Colv = FALSE, Rowv=TRUE, dendrogram= c("row"), scale ="row", labRow =rownames(pairwise.miRNA.test), labCol = colnames(pairwise.miRNA.test), main = ' \n Pairwise cluster comparison \n Significant miRNA for the Test Set', col =hmcols, cexRow =0.40,cexCol = 0.7, trace ="none", key.title = "Color Key", margins=c(6,10))
dev.off()
heatmap.2(x = pairwise.miRNA.test,Colv = FALSE, Rowv=TRUE, dendrogram= c("row"), scale ="row", labRow =rownames(pairwise.miRNA.test), labCol = colnames(pairwise.miRNA.test), main = ' \n Pairwise cluster comparison \n Significant miRNA for the Test Set', col =hmcols, cexRow =0.40,cexCol = 0.7, trace ="none", key.title = "Color Key", margins=c(6,10))
```


# Somatic Mutation

```{r upload files, eval=FALSE}
SomaticMutation_TrainingSet <- read_csv("/Volumes/GoogleDrive/My Drive/Documents/GitHub/Data/Second_data_split/TrainingSet/SomaticMutation_TrainingSet.csv")
SomaticMutation_TestSet <- read_csv("/Volumes/GoogleDrive/My Drive/Documents/GitHub/Data/Second_data_split/TestSet/SomaticMutation_TestSet.csv")
```

## Training set
```{r Chi-Square test for training set SM}
available.train<-Clinical_TrainingSet$X1 %in% SomaticMutation_TrainingSet$X1
SomaticMutation_TrainingSet$X1<-NULL
SM.c.sbc<-c.sbc[available.train]
SomaticMutation_TrainingSet$sbc.labels<-SM.c.sbc
table.SM<-list()
for (i in names(SomaticMutation_TrainingSet)){
  if (i=="sbc.labels" | length(table(SomaticMutation_TrainingSet[[i]]))<3){
    next
  }else{
  table.SM[[i]]<-chisq.test(table(SomaticMutation_TrainingSet$sbc.labels,SomaticMutation_TrainingSet[[i]]))$p.value
  }
}
tableSM.adj <- p.adjust(unlist(table.SM), method = "BH")
SM_pval <- names(tableSM.adj[tableSM.adj<0.05]) 
SM_signature<-SomaticMutation_TrainingSet[SM_pval]

par(mar=c(1,1,1,1))

SM_signature$sbc.labels<-SM.c.sbc
SM_signature<-SM_signature[order(SM_signature$sbc.labels),]
ordered_labels<-SM_signature$sbc.labels
SM_signature$sbc.labels<-NULL

```

```{r heatmap training set SM}
hmcols<-colorRampPalette(brewer.pal(11,"RdBu"))(13)
pdf(file="AnalysisOfResults_files/SomaticMutation_TrainingSet.pdf")
heatmap.2(x = t(SM_signature), Rowv =TRUE ,Colv = FALSE, dendrogram = c("row"), scale ="row",ColSideColors =c("#b16ed3","#d37f6e","#90d36e","#6ec2d3")[ordered_labels], labRow = colnames(SM_signature), labCol = NA, main = ' \n Training Set \n Significant features for Somatic Mutation Data', col =hmcols, cexRow =0.40, trace ="none", key.title = "Color Key")
legend("topright", legend = c("Good moderate","Worst","Best","Bad moderate"),fill =c("#b16ed3","#d37f6e","#90d36e","#6ec2d3"), cex = 0.4)
dev.off()

```

```{r}
hmcols<-colorRampPalette(brewer.pal(11,"RdBu"))(7)
heatmap.2(x = t(SM_signature), Rowv =TRUE ,Colv = FALSE, dendrogram = c("row"), scale ="row",ColSideColors =c("#b16ed3","#d37f6e","#90d36e","#6ec2d3")[ordered_labels], labRow = colnames(SM_signature), labCol = NA, main = ' \n Training Set \n Significant features \n for Somatic Mutation Data', col =hmcols, cexRow =0.40, trace ="none", key.title = "Color Key")
legend("topright", legend = c("Good moderate","Worst","Best","Bad moderate"),fill =c("#b16ed3","#d37f6e","#90d36e","#6ec2d3"), cex = 0.4)
```


## Test set
```{r Chi-Square test for test set SM}
available.test<-Clinical_TestSet$X1 %in% SomaticMutation_TestSet$X1
SomaticMutation_TestSet$X1<-NULL
SM.c.sbc.new<-c.sbc.new[available.test]
SomaticMutation_TestSet$sbc.labels<-SM.c.sbc.new
table.SM.new<-list()
for (i in names(SomaticMutation_TestSet)){
  if (i=="sbc.labels" | length(table(SomaticMutation_TestSet[[i]]))<3){
    next
  }else{
  table.SM.new[[i]]<-chisq.test(table(SomaticMutation_TestSet$sbc.labels,SomaticMutation_TestSet[[i]]))$p.value
  }
}
tableSM.adj.new <- p.adjust(unlist(table.SM.new), method = "BH")
SM_pval.new <- names(tableSM.adj.new[order(tableSM.adj.new)[1:50]]) 
SM_signature.new<-SomaticMutation_TestSet[SM_pval.new]

par(mar=c(1,1,1,1))

SM_signature.new$sbc.labels<-SM.c.sbc.new
SM_signature.new<-SM_signature.new[order(SM_signature.new$sbc.labels),]
ordered_labels.new<-SM_signature.new$sbc.labels
SM_signature.new$sbc.labels<-NULL

```

```{r heatmap test set SM}
hmcols<-colorRampPalette(brewer.pal(11,"RdBu"))(13)
heatmap.2(x = t(SM_signature.new), Rowv =TRUE ,Colv = FALSE, dendrogram = c("row"), scale ="row",ColSideColors =c("#b16ed3","#d37f6e","#90d36e","#6ec2d3")[ordered_labels.new], labRow = colnames(SM_signature.new), labCol = NA, main = ' \n Test Set \n Significant features \n for Somatic Mutation Data', col =hmcols, cexRow =0.40, trace ="none", key.title = "Color Key")
legend("topright", legend = c("Good moderate","Worst","Best","Bad moderate"),fill =c("#b16ed3","#d37f6e","#90d36e","#6ec2d3"), cex = 0.4)
```

# Clinical Data

## Training Set
```{r Contingency analysis for clinical variables in Training Set }
Clinical_TrainingSet$sbc.labels<-c.sbc
for (i in colnames(Clinical_TrainingSet)){
  if (i=="X1"|i=="sbc.labels"){
    next
  }else{
    dummie<-chisq.test(table(Clinical_TrainingSet[[i]],Clinical_TrainingSet$sbc.labels))
    if (is.nan(dummie$p.value)==FALSE && dummie$p.value<0.05){
      print(i)
      print(dummie)
    }
  }
}

```

## Test Set
```{r Contingency analysis for clinical variables in Test Set}
Clinical_TestSet$sbc.labels<-c.sbc.new
for (i in colnames(Clinical_TestSet)){
  if (i=="X1"|i=="sbc.labels"){
    next
  }else{
    dummie<-chisq.test(table(Clinical_TestSet[[i]],Clinical_TestSet$sbc.labels))
    if (is.nan(dummie$p.value)==FALSE && dummie$p.value<0.05){
      print(i)
      print(dummie)
    }
  }
}
```

# Posterior probability plot

```{r}
pdf(file="AnalysisOfResults_files/PosteriorProbability.pdf")
par(mar=c(1,1,1,1))
heatmapdata <- as.data.frame(final.betahat)
heatmap.2(t(as.matrix(heatmapdata)),dendrogram="none", margins=c(6,10), main = "Posterior prob. \n for Selection \n ", cexCol = 0.85, cexRow = 0.7, Rowv = FALSE)
dev.off()
```

```{r}
par(mar=c(1,1,1,1))
heatmapdata <- as.data.frame(final.betahat)
heatmap.2(t(as.matrix(heatmapdata)),dendrogram="none", margins=c(6,10), main = "Posterior prob. \n for Selection \n ", cexCol = 0.85, cexRow = 0.7, Rowv = FALSE)
```

